### PR TITLE
Mlperf models resnet50 and mobilenetv1 support

### PR DESCRIPTION
**Pull request by kahmed10**
Added necessary changes allowing MIGraphX to run protobuf and onnx files for these models in MLPerf. I also added tests for variable batch size (and found a duplicate unit test, removed in PR).

### CODE REVIEW

File: src/onnx/onnx.cpp [hunk 1]
```diff
@@ -1686,13 +1687,15 @@ struct onnx_parser
...
                            {
-                               long default_batch_size = 1; // FIXME
-                               return default_batch_size;
+                               if(static_cast<int>(d.dim_value()) <= 0)
+                                   return batch_size;
+                               return d.dim_value();
                            }
-                           return d.dim_value();
+                           return batch_size;
+                           //    return d.has_dim_value() ? d.dim_value() : batch_size;
```
Comment (kahmed10) [inline src/onnx/onnx.cpp]: will remove this comment when I get back.

File: src/tf/tf.cpp [hunk 2]
```diff
@@ -189,6 +190,7 @@ struct tf_parser
         add_binary_op("SquaredDifference", op::sqdiff{});
         add_binary_op("Sub", op::sub{});
 
+        add_mem_op("ArgMax", &tf_parser::parse_arg_op<op::argmax>, false);
```
[the remark this answers is not in the document]
Comment (kahmed10) [reply src/tf/tf.cpp]: fixed

File: doc/src/reference/py.rst [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+    :param str filename: Path to file.
+    :param str batch_size: default batch size to use (if not specified in onnx file).
+
+    :rtype: program
+
+parse_tf
+----------
+
+.. py:function:: parse_tf(filename, is_nhwc=True)
```
Comment (pfultz2) [inline doc/src/reference/py.rst]: `batch_size` should be listed here.

File: doc/src/reference/py.rst [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    :param calibration: Calibration data used to decide the parameters to the int8 optimization.
+    :type calibration: list[dict[str, argument]]
+    :param ins_names: List of instructions to quantize.
+    :type ins_names: list[str]
+
+
+parse_onnx
+----------
+
+.. py:function:: parse_onnx(filename)
```
Comment (pfultz2) [inline doc/src/reference/py.rst]: `batch_size` should be listed here with its default value.

File: src/include/migraphx/tf.hpp [hunk 5]
```diff
@@ -7,8 +7,15 @@
...
+/// struct to pass in tf options to parser
+struct tf_options
+{
+    bool is_nhwc            = false;
+    unsigned int batch_size = 1;
+};
+
 /// Create a program from a tf pb file (default is nhwc format)
-program parse_tf(const std::string& name, bool is_nhwc);
+program parse_tf(const std::string& name, tf_options options);
```
Comment (pfultz2) [inline src/include/migraphx/tf.hpp]: The `options` struct should be optional. It should be `parse_tf(const std::string& name, tf_options options=tf_options{})`

File: src/include/migraphx/onnx.hpp [hunk 6]
```diff
@@ -7,8 +7,14 @@
...
 
+/// struct to pass in onnx options to parser
+struct onnx_options
+{
+    unsigned int batch_size = 1;
+};
+
 /// Create a program from an onnx file
-program parse_onnx(const std::string& name);
+program parse_onnx(const std::string& name, onnx_options options);
```
Comment (pfultz2) [inline src/include/migraphx/onnx.hpp]: The `options` should be optional. It should be `parse_onnx(const std::string& name, onnx_options options=onnx_options{})`.

File: src/onnx/cifar10.cpp [hunk 7]
```diff
@@ -53,7 +53,7 @@ int main(int argc, char const* argv[])
     std::string gpu_cpu  = argv[1];
     std::string file     = argv[2];
     std::string datafile = argv[3];
-    auto prog            = migraphx::parse_onnx(file);
+    auto prog            = migraphx::parse_onnx(file, migraphx::onnx_options{});
```
Comment (pfultz2) [inline src/onnx/cifar10.cpp]: This shouldn't be changed.

File: src/onnx/mnist.cpp [hunk 8]
```diff
@@ -118,7 +118,7 @@ int main(int argc, char const* argv[])
         std::vector<int32_t> labels = read_mnist_labels(labelfile, nlabels);
 
         std::string file = argv[1];
-        auto prog        = migraphx::parse_onnx(file);
+        auto prog        = migraphx::parse_onnx(file, migraphx::onnx_options{});
```
Comment (pfultz2) [inline src/onnx/mnist.cpp]: This shouldn't be changed.

**Comment by kahmed10** [comment]
Fixed suggestions @pfultz2
